helpers/javascript/lib/updater.js

```diff
@@ -0,0 +1,91 @@
+/* DEPENDENCY FILE UPDATER
+ *
+ * Inputs:
+ *  - directory containing a package.json and a yarn.lock
+ *  - dependency name
+ *  - new dependency version
+ *
+ * Outputs:
+ *  - updated package.json and yarn.lock files
+ *
+ * Update the dependency to the version specified and rewrite the package.json
+ * and yarn.lock files.
+ */
+const fs = require("fs");
+const path = require("path");
+const { Add } = require("yarn/lib/cli/commands/add");
+const Config = require("yarn/lib/config").default;
+const { NoopReporter } = require("yarn/lib/reporters");
+const Lockfile = require("yarn/lib/lockfile/wrapper").default;
+
+// Add is a subclass of the Install CLI command, which is responsible for
+// adding packages to a package.json and yarn.lock. Upgrading a package is
+// exactly the same as adding, except the package already exists in the
+// manifests.
+//
+// Usually, calling Add.init() would execute a series of steps: resolve, fetch,
+// link, run lifecycle scripts, cleanup, then save new manifest (package.json).
+// We only care about the first and last steps: resolve, then save the new
+// manifest. Fotunately, overriding bailout() gives us an opportunity to skip
+// over the intermediate steps in a relatively painless fashion.
+class LightweightAdd extends Add {
+  async bailout(patterns) {
```

Comment:
My lack of familiarity with the internals of `yarn` makes me want to see more context about `bailout` here, specifically how it fits into the Add flow of execution? @hmarr thoughts?